grpcutil/jwt_auth.go

```diff
@@ -39,22 +39,39 @@ type JWTAuthenticationHandler struct {
 	// the jwt ourselves. This flag skips the validation, and instead trusts the JWT has been validated
 	// upstream.
 	ValidationDisabled bool
+	// Endpoint is the endpoint where JWKs public keys can be found to do JWT validation.
+	Endpoint string
 }
 
 // NewJWTAuthenticationHandler returns a JWTAuthenticationHandler with a verifier initialized. Uses defaults
 // for JWT related fields that will retreive a user email when using IAM on GCP.
-func NewJWTAuthenticationHandler(ctx context.Context, endpoint string, disableValidation bool) (*JWTAuthenticationHandler, error) {
-	verifier, err := jwtutil.NewVerifier(ctx, endpoint)
-	if err != nil {
-		return nil, err
-	}
-	return &JWTAuthenticationHandler{
-		Verifier:           verifier,
+func NewJWTAuthenticationHandler(ctx context.Context, opts ...HandlerOption) (*JWTAuthenticationHandler, error) {
```

Comment:
```suggestion
func NewJWTAuthenticationHandler(ctx context.Context, opts ...JWTAuthOption) (*JWTAuthenticationHandler, error) {
```

I recommend rename the option type to be more specific to this handler. This would help avoid future naming conflicts with other utils in the package.